data/tests/test.js

```diff
@@ -82,6 +82,29 @@ axios.get(flagURL).then((response) => {
     console.error(error);
 });
 
-
-
-
+function roundTo(num, nearest = 1){
+    return Math.round(num/nearest)*nearest;
+}
+
+console.log("Areas Consistency Test");
+var numberOfCountriesWithCorrectAreas = 0;
+var totalCountries = countriesJSON.length;
+for(var i = 0; i < totalCountries; i++){
+    var countryJSON = countriesJSON[i];
+    var areaInKm2 = countryJSON.area.km2;
+    var areaInMi2 = countryJSON.area.mi2;
+    var expectedAreaInMi2 = 0.386102*areaInKm2;
+    var roundingPlace = -3;
+    var roundedAreaInMi2 = roundTo(expectedAreaInMi2, Math.pow(10, roundingPlace));
+    while(roundedAreaInMi2){
+      var diff = Math.abs(roundedAreaInMi2 - areaInMi2);
+      if(diff<=1){
+          numberOfCountriesWithCorrectAreas += 1;
+          break;
+      }
+      roundingPlace += 1;
+      roundedAreaInMi2 = roundTo(expectedAreaInMi2, Math.pow(10, roundingPlace));
+    }
+}
+console.log(`Consistent: ${numberOfCountriesWithCorrectAreas}`);
+console.log(`Total: ${totalCountries}\n`);
```

Comment:
How did you run this to do the testing? To execute all tests that are in the project, you need to run `npm test` in _terminal_. The tests are located in `spec/getCountriesSpec.js`.
Also, I don't understand what is this code doing here, it looks pretty complex tho.

What we need to do in the test is:
1. Describe a test
2. Get all countries
3. Iterate through each `Country {Object}` case
    - Get `area.km2` and `area.mi2`
    - Convert one to another unit
    - Round units to nearest integer
    - Compare whether values are equal

Reply:
### Details about test
As per the data I found in Wikipedia, some countries' area in mi<sup>2</sup> was rounded to nearest 1000 or 100 (not just the nearest integer). So this test rounds the calculated area until it matches.

Also, this tests disregards an error of ±1 in area.

### Looking for suggestions
I was also wondering why does this `test.js` exist.
Testing of functions is done in `spec/`.

So, I assumed that this file was meant for testing the consistency of the data in `data.json`. Also, the test for flags in `test.js` is not 100% passing.


What should be done? 🤔

Reply:
> As per the data I found in Wikipedia, some countries' area in mi2 was rounded to nearest 1000 or 100 (not just the nearest integer). So this test rounds the calculated area until it matches.

All we need (at least for now), is to have as accurate and precise value as possible. If someone notice that some values are incorrect for some country, we can adjust it.

Important thing is that we should be consistent with rounding and it should work properly for small and big countries (like _Vatican_ 🇻🇦 and _Russia_ 🇷🇺)

What do you think @bhatvikrant, how should we resolve this?

> I was also wondering why does this test.js exist.

Me neither 🤷‍♂️ For now, just ignore this.

---

What would be really great is if you would like to prepare a `getCountriesByAreaRange(from, to)` Method and write tests for it (all tests are located in `spec/getCountriesSpec.js`). After you write a test, you can execute it by running in terminal `npm test` Script.

Reply:
> All we need (at least for now), is to have as accurate and precise value as possible.

Since the data have been copy-pasted from Wikipedia, it would be better to stick to these values instead of changing them ourselves. We can't see anything better than crowd-wisdom of Wikipedia. 🙂

As for the function `getCountriesByAreaRange(from, to)`, I'd open an issue and see if someone would like to take it up.

@bhatvikrant What do you think?